juju_verify/exceptions.py

```diff
@@ -24,3 +29,17 @@ class VerificationError(Exception):
 
 class CharmException(Exception):
     """Exception related to the charm or the unit that runs it."""
+
+
+class JujuActionFailed(Exception):
+    """Exception related to failing action run on the unit."""
+
+    def __init__(self, error: JujuError, unit: Unit, action: str,
+                 params: Optional[Dict[str, Any]] = None):
+        """Initialize ActionFailed error message."""
```

Comment:
Sorry, in this docstring, `ActionFailed` should probably be renamed to `JujuActionFailed`, too.